addons/hr_attendance/models/hr_employee.py

```diff
@@ -53,10 +53,10 @@ def attendance_scan(self, barcode):
     @api.multi
     def attendance_manual(self, next_action, entered_pin=None):
         self.ensure_one()
-        if not (entered_pin is None) or self.env.user.has_group('hr_attendance.group_hr_attendance_use_pin') and (self.user_id and self.user_id == self.env.user):
-            if entered_pin != self.pin:
-                return {'warning': _('Wrong PIN')}
-        return self._attendance_action(next_action)
+        can_check_without_pin = self.user_id == self.env.user and entered_pin is None
+        if can_check_without_pin or self.env.user.has_group('hr_attendance.group_hr_attendance_use_pin') and entered_pin is not None and entered_pin == self.sudo().pin:
```

Comment:
Simple employee cannot checkin if user does not have `hr_attendance.group_hr_attendance_use_pin`
This group should be removed, it does not make sense to change the behavior for all employees only based on the group of the logged in user

Reply:
But this group should be given to the user via the config parameter, perhaps we should do as it was done before and check that on the SUPERUSER_ID instead of the logged in user.
In that case, we should put a comment as to why we check it on the SUPERUSER_ID.